validator/sawtooth_validator/networking/interconnect.py

```diff
@@ -402,14 +402,19 @@ def send(self, message_type, data, connection_id, has_callback=False):
                 message_type=message_type)
 
             fut = future.Future(message.correlation_id, message.content,
-                                has_callback=has_callback)
+                                has_callback=True if callback is not None
```

Comment:
This is how the future implementation currently works. I agree just passing the callback would be easier, but that seems like a separate story.